lib/debugger/debugger.lua

```diff
@@ -268,7 +263,15 @@ function Debugger:autoInitialize(loop)
 
 			loop.profiling = loop.profiling or {}
 
-			if eventName == self._eventOrder[1] then
+			if self._currentFrame[eventName] then
+				self.plasma.finishFrame(plasmaNode)
```

Comment:
Calling finishFrame at the start of the frame instead of the end of the frame means that UI elements that are supposed to be removed will still exist for an additional frame. This could cause unintended code to run and cause flashing. 

For example, in one frame the debugger might replace an item in a list. The proposed changed would mean that both items, new and old, would be visible for exactly one frame before the old one is removed. This is not ideal.